api/envoy/extensions/router/cluster_specifier/lua/v3/lua.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package envoy.extensions.router.cluster_specifier.lua.v3;
```

Comment:
I think this should an extension point of `route` rather than `router`. And the `cluster_specifier` should be plural form.

Reply:
https://github.com/envoyproxy/envoy/blob/acc54c548214504a17704e75bf7e6a141c02ae8f/envoy/router/cluster_specifier_plugin.h#L18

the interface class for creating cluster specifier plugin is just in `router`.